lib/kernel/src/inet_dns.erl

```diff
@@ -246,9 +254,38 @@ decode_rr_section(Bin, N, Buffer, RRs) ->
                           z                = Z,
                           data             = D,
                           do               = DnssecOk};
+                   ?S_TSIG ->
+                       % RFC 8945: 5.2. FORMERR if not last
+                       Rest =/= <<>> andalso throw(?DECODE_ERROR),
+                       % RFC 8945: 5.2. FORMERR if more than one dns_rr_tsig
+                       lists:keymember(dns_rr_tsig, 1, RRs) andalso
+                        throw(?DECODE_ERROR),
```

Comment:
Isn't this redundant, since the only possibility for `?S_TSIG` to occur more than once is to not be last?

Reply:
Yep.

Resolved by https://github.com/erlang/otp/pull/6985/commits/d3e5974876f4a1f3fb9a356731b10ff806076846